core/mm/core_mmu.c

```diff
@@ -2451,16 +2451,20 @@ vaddr_t core_mmu_get_va(paddr_t pa, enum teecore_memtypes type, size_t len)
 }
 
 #ifdef CFG_WITH_PAGER
-bool is_unpaged(void *va)
+bool is_unpaged(const void *va)
 {
 	vaddr_t v = (vaddr_t)va;
 
 	return v >= VCORE_START_VA && v < get_linear_map_end_va();
 }
-#else
-bool is_unpaged(void *va __unused)
+#endif
+
+#ifdef CFG_NS_VIRTUALIZATION
+bool is_nexus(const void *va)
 {
-	return true;
+	vaddr_t v = (vaddr_t)va;
+
+	return v >= VCORE_START_VA && v < VCORE_NEX_RW_PA + VCORE_NEX_RW_SZ;
```

Comment:
This `_PA` in `VCORE_NEX_RW_PA` is a bit disturbing. Should all the `_PA` and `_VA` defines in `linker.h` be renamed `_ADDR`? They reference symbols that may be relocated so depending on whether or not the MMU is turned on, they may be PAs or VAs, and when they are VAs their value may differ from the PAs when ASLR is enabled.

Reply:
Yes, it would be nice with a consistent naming for address defines that are updated with relocation (ASLR and `CFG_CORE_PHYS_RELOCATABLE=y`).
We should do something similar for the address defines that don't change with relocation.

But I'd rather do it in a separate PR.